feat(ui-ux): allow dusd loops in vaults

lykalabrada opened the pull request:
<!--  Thanks for sending a pull request! -->

#### What this PR does / why we need it:
In line with ain changes (https://github.com/DeFiCh/ain/pull/1971), DUSD loops in vaults should now be allowed

#### Which issue(s) does this PR fixes?:

<!--
(Optional) Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

Fixes #4034, DFC-394

#### Additional comments?:
When testing this PR, please check the following scenarios:
- Users should be able to take a DUSD loan when their vault has the following collaterals:
  - Vault that has 100% DFI collateral
  - Vault that 100% DUSD collateral (_new_)
  - 50% DFI + other tokens

- Users should NOT be able to take a DUSD loan when their vault has the following collaterals:
  - 0% DFI + other tokens
  - <50% DFI + other tokens
  - DUSD + other tokens

- Users should NOT be able to take a **non-DUSD** loan when their vault has the following collaterals:
  - 100% DUSD
  -  <50% DFI + other tokens

- Others:
  - Removing collateral from a vault that has an active DUSD loan -> verify that this is not allowed
  - Adding of non-DUSD collateral to a vault with an active DUSD loan, containing 100% DUSD collateral -> verify that this is not allowed


#### Developer Checklist:

<!--
Merging into the main branch implies your code is ready for production.
Before requesting for code review, please ensure that the following tasks
are completed. Otherwise, keep the PR drafted.
-->

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

<!--
* If applicable
-->

Krysh90 commented on mobile-app/app/screens/AppNavigator/screens/Loans/screens/AddOrRemoveCollateralScreen.tsx:
```diff
@@ -361,54 +361,81 @@ export function AddOrRemoveCollateralScreen({ route }: Props): JSX.Element {
+      if (
+        !["DFI", "DUSD"].includes(selectedCollateralItem.token.displaySymbol) ||
+        (selectedCollateralItem.token.displaySymbol === "DFI" &&
+          isDFILessThanHalfOfRequiredCollateral)
```
I guess here is a small issue, if I have a vault with DFI + DUSD as collateral and I take a small DUSD loan which is below that 50% DFI requirement. and then try to take out all DFI while staying above the collateral ratio. I would have still a valid vault because only DUSD are in as collateral.

lykalabrada commented:
> One thing came to my mind regarding your comment about my PR. You absolutely correct that adding/removing collateral was missing. I found one case, where it was especially needed. If you have a vault with DFI + DUSD as collateral and DUSD as loan. If you then try to remove all the DFI while still having enough DUSD to cover the loan, it should be possible to take out the DFI.
> 
> As before the 50% DFI requirement was valid and afterwards the 100% DUSD. I added a comment to the PR, where I think this check is missing. As it is not working on your current feature branch.
> 
> I have tested a few of my tests, which all passed.

Thanks for checking! You are correct, I updated the PR to enable the removal of all DFI in the presence of only DUSD.

fullstackninja864 commented on mobile-app/app/screens/AppNavigator/screens/Loans/hooks/ValidateDUSDLoanAndCollateral.ts:
```diff
@@ -0,0 +1,74 @@
+import { useSelector } from "react-redux";
+import { getActivePrice } from "../../Auctions/helpers/ActivePrice";
+
+interface useValidateDUSDLoanAndCollateralProps {
```
Lets use PascalCase for interface name

fullstackninja864 commented on mobile-app/app/screens/AppNavigator/screens/Loans/hooks/ValidateLoanAndCollateral.ts:
```diff
@@ -0,0 +1,76 @@
+      totalRequiredCollateral.dividedBy(2),
+    );
+
+  const isDUSD100PercentOfCollateral = props.collateralAmounts.every(
```
```suggestion
  // check if all collaterals is of DUSD token
  const isDUSD100PercentOfCollateral = props.collateralAmounts.every(
```

fullstackninja864 commented on shared/translations/languages/de.json:
```diff
@@ -1772,7 +1772,7 @@
     "Vault does not meet min. collateral ratio": "Vault erfüllt die Mindestbesicherungsquote nicht",
     "Insufficient DFI and/or DUSD in vault to maintain active loans": "Unzureichende DFI und/oder DUSD im Vault zur Aufrechterhaltung aktiver Darlehen",
-    "Active DUSD loans require 50% DFI collaterals": "Aktive DUSD-Darlehen erfordern 50 % DFI-Besicherung",
+    "Active DUSD loans require 50% DFI or 100% DUSD collaterals": "Aktive DUSD-Darlehen erfordern 50% DFI- oder 100% DUSD-Sicherheiten",
```
whan `isFeatureAvailable("loop_dusd")` will be false in that case we should show `50% DFI or 100% DUSD collaterals` ?
